safehttp/mux.go

```diff
@@ -92,35 +89,41 @@ func NewServeMux(d Dispatcher, domains ...string) *ServeMux {
 		dm[host] = true
 	}
 	return &ServeMux{
-		mux:       http.NewServeMux(),
-		domains:   dm,
-		disp:      d,
-		handlers:  map[string]methodHandler{},
-		interceps: map[string]Interceptor{},
+		mux:      http.NewServeMux(),
+		domains:  dm,
+		disp:     d,
+		handlers: map[string]methodHandler{},
 	}
 }
 
+type appliedInterceptor struct {
+	it  Interceptor
+	cfg Config
+}
+
 // Handle registers a handler for the given pattern and method. If another
 // handler is already registered for the same pattern and method, Handle panics.
 //
 // Configs can be optionally passed in order to modify the behavior of the
 // interceptors on a registered handler. Passing a Config whose corresponding
-// Interceptor was not installed will produce no effect.
+// Interceptor was not installed will produce no effect. If multiple Configs are
+// passed for the same Interceptor, only the first one will take effect.
 func (m *ServeMux) Handle(pattern string, method string, h Handler, cfgs ...Config) {
-	cfgMap := map[string]Config{}
-	for _, cfg := range cfgs {
-		for k, it := range m.interceps {
-			if cfg.Match(it) {
-				cfgMap[k] = cfg
+	var interceps []appliedInterceptor
+	for _, it := range m.interceps {
+		var cfg Config
+		for _, c := range cfgs {
+			if c.Match(it) {
+				cfg = c
+				break
 			}
 		}
```

Comment:
Could you file a feature request to ban adding two configurations to the same interceptor?

Reply:
For posterity: https://github.com/google/go-safeweb/issues/148.